adapters/resetdigital/resetdigital.go

```diff
@@ -4,305 +4,237 @@ import (
-	headers := http.Header{}
-
-	addNonEmptyHeaders(&headers, map[string]string{
-		"Content-Type": "application/json;charset=utf-8",
-		"Accept":       "application/json",
-	})
-
-	if request != nil && request.Device != nil {
-		addNonEmptyHeaders(&headers, map[string]string{
-			"Accept-Language": request.Device.Language,
-			"User-Agent":      request.Device.UA,
-			"X-Forwarded-For": request.Device.IP,
-			"X-Real-Ip":       request.Device.IP,
-		})
-	}
-	if request != nil && request.Site != nil {
-		addNonEmptyHeaders(&headers, map[string]string{
-			"Referer": request.Site.Page,
-		})
-	}
-
-	return headers
+type adapter struct {
+	endpoint string
 }
 
-func addNonEmptyHeaders(headers *http.Header, headerValues map[string]string) {
-	for key, value := range headerValues {
-		if len(value) > 0 {
-			headers.Add(key, value)
-		}
-	}
+func Builder(_ openrtb_ext.BidderName, cfg config.Adapter, _ config.Server) (adapters.Bidder, error) {
+	return &adapter{
+		endpoint: cfg.Endpoint,
+	}, nil
 }
 
-func (a *adapter) MakeRequests(requestData *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
-	var (
-		requests []*adapters.RequestData
-		errors   []error
-	)
-
-	for _, imp := range requestData.Imp {
-		bidType, err := getBidType(imp)
-		if err != nil {
-			errors = append(errors, err)
-			continue
-		}
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	_ = reqInfo
 
-		splittedRequestData, err := processDataFromRequest(requestData, imp, bidType)
-		if err != nil {
-			errors = append(errors, err)
-			continue
-		}
-
-		requestBody, err := json.Marshal(splittedRequestData)
+	if len(request.Imp) != 1 {
+		return nil, []error{&errortypes.BadInput{
+			Message: "ResetDigital adapter supports only one impression per request",
+		}}
+	}
 
-		if err != nil {
-			errors = append(errors, err)
-			continue
-		}
+	errs := make([]error, 0, 1)
 
-		requests = append(requests, &adapters.RequestData{
-			Method:  "POST",
-			Uri:     a.endpointUri,
-			Body:    requestBody,
-			Headers: getHeaders(requestData),
-			ImpIDs:  []string{imp.ID},
-		})
+	imp := request.Imp[0]
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Error parsing bidderExt from imp.ext: %v", err),
+		}}
 	}
 
-	return requests, errors
-}
+	var resetDigitalExt openrtb_ext.ImpExtResetDigital
+	if err := json.Unmarshal(bidderExt.Bidder, &resetDigitalExt); err != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Error parsing resetDigitalExt from bidderExt.bidder: %v", err),
+		}}
+	}
 
-func processDataFromRequest(requestData *openrtb2.BidRequest, imp openrtb2.Imp, bidType openrtb_ext.BidType) (resetDigitalRequest, error) {
-	var reqData resetDigitalRequest
+	if resetDigitalExt.PlacementID == "" {
```

Comment:
The `"minLength": 1`, and `"required": ["placement_id"],` entries in `static/bidder-params/resetdigital.json` make the true branch of this `if` statement, in practice, unreachable. I believe it is safe if we remove the check if you want